Add hooks for ESLint and Stylelint. Update Rubocop hook

PuercoWork opened the pull request:
1. Fix for when running 
2. Add a hook for ESLint and for Stylelint.

There are other variations of git hooks. One running sass through prettier, another one running prettier as a stand-alone.

Commit:
Rubocup hook: Exit if there are not files to check

If RuboCop::TargetFinder.new is invoked with empty args it recursively
find all Ruby source files under the current directory. We do not want
that behaviour

Commit:
Add pre-commit hooks for ESLint and Stylelint

beathyate commented on lib/able_git_hooks/generators/templates/rubocop.rb:
```diff
@@ -4,6 +4,9 @@
 
 staged_files = `git diff -z --staged --name-only --diff-filter=ACM`.split("\u0000")
+
+exit 0 unless staged_files.empty?
```
Shouldn't this be an `if`?

PuercoWork replied:
Indeed, I was using `.any?` in the hook submitted to WTA and I forgot to change to if here

beathyate reviewed (changes requested):
You also need to add the methods on `install_generator.rb` and the tests.

Commit:
Update Rubocop hook test

The file is now named rubocop.rb and doesn't contain the string bundle
exec.

Commit:
_do_hook: Use eval instead of exec

exec creates a sub-process to evaluate the hook in while eval
doesn't. Creating a subprocess is costly and I don't see a reason to
prefer it in this case.

h/t Marsam

PuercoWork commented:
@goddamnhippie I've included tests, updated the rubocop hook test which was failing and added default configs for ESLint and prettier.

The thing that is remaining is to don't include the stylelint hook and config by default unless a CLI option is passed

beathyate commented:
@PuercoWork sounds good, do you want me to add the option?

PuercoWork commented:
@goddamnhippie I've added the CLI option since, although I've not written a test case for when the option is true

Commit:
Add Rubocop config

beathyate commented on lib/able_git_hooks/generators/install_generator.rb:
```diff
@@ -22,6 +25,33 @@ def install_rubocop_hooks
+      end
+
+      def install_stylelint_hooks
+        puts "options: ", options
```
I think you forgot to delete this.

beathyate commented on lib/able_git_hooks/generators/install_generator.rb:
```diff
@@ -22,6 +25,33 @@ def install_rubocop_hooks
+
+      def install_stylelint_hooks
+        puts "options: ", options
+        if options['with_stylelint']
```
Rubocop will complain about ' vs "

PuercoWork replied:
/me nods

I've included the rubocop.yml and enabled rubocop in my editor to prevent this style issues for occurring in the future. There where also a couple of trailing whitespace in the buffer

beathyate commented on lib/able_git_hooks/generators/install_generator.rb:
```diff
@@ -22,6 +25,33 @@ def install_rubocop_hooks
+      end
+
+      def copy_stylelint_defaults
+        if options['with_stylelint']
```
Same here ' vs "

Commit:
Add test for when the --with-stylelint option is passed

How the current test suite is structured is that the generator is part
of the setup and each test method is an assertion. Respecting that
structure, when the generator is going to be invoked with different
options I used a different test class.

Another alternative would be to move the call to run_generator to each
test_* method., but the current structure favors that each test method
tests only one thing.

PuercoWork commented:
@goddamnhippie I've included the test case for the `--with-stylelint` option, let me know what you think of having it in a separate test class
